File: app/src/main/java/com/shuneault/netrunnerdeckbuilder/game/CycleList.java
package com.shuneault.netrunnerdeckbuilder.game;

import java.util.ArrayList;

public class CycleList extends ArrayList<Cycle> {

    public Cycle getCycle(String cycleCode) {
        for (Cycle cycle : this)
            if (cycle.getCode().equals(cycleCode))
                return cycle;

        return null;
    }
}

File: app/src/main/java/com/shuneault/netrunnerdeckbuilder/game/CycleList.kt
package com.shuneault.netrunnerdeckbuilder.game
import java.util.ArrayList

class CycleList : ArrayList<Cycle>() {
    fun getCycle(cycleCode: String): Cycle? {
        for (cycle in this) if (cycle.code == cycleCode) return cycle
        return null
    }
}